### PR TITLE
Debt: avoid promise.cancel in zip.ts

### PR DESCRIPTION
**Pull request by bpasero**
For https://github.com/Microsoft/vscode/issues/56137

@sandy081 adding you because this area is sensitive for extracting extensions. I converted the code to use `CancellationToken` and avoid `TPromise.cancel()`. 

/cc @jrieken

### CODE REVIEW

File: src/vs/base/node/zip.ts [hunk 1]
```diff
@@ -78,72 +80,88 @@ function extractEntry(stream: Readable, fileName: string, mode: number, targetPa
...
 	let isCanceled = false;
-	let last = TPromise.wrap<any>(null);
+	let last = createCancelablePromise(() => Promise.resolve(null));
 	let extractedEntriesCount = 0;
 
-	return new TPromise((c, e) => {
-		const throttler = new SimpleThrottler();
+	return createCancelablePromise(token => {
+
+		once(token.onCancellationRequested)(() => {
```
Comment (jrieken) [inline src/vs/base/node/zip.ts]: No `once` not needed as the event will only ever fire once

File: src/vs/base/node/zip.ts [hunk 2]
```diff
@@ -78,72 +80,88 @@ function extractEntry(stream: Readable, fileName: string, mode: number, targetPa
...
-		}, e));
-		zipfile.readEntry();
-		zipfile.on('entry', (entry: Entry) => {
-
-			if (isCanceled) {
-				return;
-			}
-
-			if (!options.sourcePathRegex.test(entry.fileName)) {
-				readNextEntry();
```
Comment (sandy081) [inline src/vs/base/node/zip.ts]: Why are we not passing the real token instead of none token?

File: src/vs/base/node/zip.ts [hunk 3]
```diff
@@ -78,72 +80,88 @@ function extractEntry(stream: Readable, fileName: string, mode: number, targetPa
...
-
-			last = throttler.queue(() => stream.then(stream => extractEntry(stream, fileName, mode, targetPath, options).then(() => readNextEntry())));
+			zipfile.on('entry', (entry: Entry) => {
+
+				if (isCanceled) {
+					return;
+				}
+
+				if (!options.sourcePathRegex.test(entry.fileName)) {
+					readNextEntry(CancellationToken.None);
```
Comment (sandy081) [inline src/vs/base/node/zip.ts]: Why are we not passing the real token instead of none token?

Comment (bpasero) [reply src/vs/base/node/zip.ts]: Good catch, I guess I added this first and only later added the token.

**Comment by bpasero** [comment]
@sandy081 can you check again, I also removed `isCanceled` with `token.isCancelationRequested`

**Comment by sandy081** [comment]
LGTM

**Pull request merged**